homeassistant/components/vodafone_station/helpers.py

```diff
@@ -0,0 +1,67 @@
+"""Vodafone Station helpers."""
+
+from typing import Any
+
+from homeassistant.components.device_tracker import DOMAIN as DEVICE_TRACKER_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from .const import _LOGGER
+
+
+async def cleanup_device_tracker(
+    hass: HomeAssistant, config_entry: ConfigEntry, devices: dict[str, Any]
+) -> None:
+    """Cleanup stale device tracker."""
+    entity_reg: er.EntityRegistry = er.async_get(hass)
+
+    ha_entity_reg_list: list[er.RegistryEntry] = er.async_entries_for_config_entry(
```

Comment:
```suggestion
    ha_entity_reg_list = er.async_entries_for_config_entry(
```

`async_entries_for_config_entry` is already typed so the hint is redundant here.

Reply:
It looks like its only used one place. I'd drop the intermediate var as it makes it harder to follow
```diff
diff --git a/homeassistant/components/vodafone_station/helpers.py b/homeassistant/components/vodafone_station/helpers.py
index 7266a215448..ac1ee09f01f 100644
--- a/homeassistant/components/vodafone_station/helpers.py
+++ b/homeassistant/components/vodafone_station/helpers.py
@@ -15,10 +15,6 @@ async def cleanup_device_tracker(
 ) -> None:
     """Cleanup stale device tracker."""
     entity_reg: er.EntityRegistry = er.async_get(hass)
-
-    ha_entity_reg_list: list[er.RegistryEntry] = er.async_entries_for_config_entry(
-        entity_reg, config_entry.entry_id
-    )
     entities_removed: bool = False
 
     device_hosts_macs = set()
@@ -27,7 +23,9 @@ async def cleanup_device_tracker(
         device_hosts_macs.add(mac)
         device_hosts_names.add(device_info.device.name)
 
-    for entry in ha_entity_reg_list:
+    for entry in er.async_entries_for_config_entry(
+        entity_reg, config_entry.entry_id
+    ):
         if entry.domain != DEVICE_TRACKER_DOMAIN:
             continue
         entry_name = entry.name or entry.original_name
```

Reply:
Applied